.github/workflows/python.yml

```diff
@@ -75,22 +81,15 @@ jobs:
         env:
           JOB_CONTEXT: ${{ toJson(job) }}
         run: echo "$JOB_CONTEXT"
-      # Sets TAGGED_OR_MAIN if this workflow is running on a tag or the main branch.
-      - name: Set TAGGED_OR_MAIN
-        if: startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main'
-        run: echo "TAGGED_OR_MAIN=1" >> $GITHUB_ENV
 
       - id: set-matrix
         shell: bash
         run: |
           matrix=()
-
-          if [[ $TAGGED_OR_MAIN ]]; then
-            # MacOS build is really slow (>30 mins); uses up a lot of CI minutes
-            matrix+=('{"platform": "macos", "runs_on": "macos-latest"},')
-          fi
-          matrix+=('{"platform": "windows", "runs_on": "windows-latest-8-cores"},')
-          matrix+=('{"platform": "linux", "runs_on": "ubuntu-latest-16-cores", container: {"image": "rerunio/ci_docker:0.5"}}')
+          matrix+=('{"platform": "macos", "target": "x86_64-apple-darwin", "runs_on": "macos-latest"},')
+          matrix+=('{"platform": "macos", "target": "aarch64-apple-darwin", "runs_on": "macos-latest"},')
+          matrix+=('{"platform": "windows", "target": "x86_64-pc-windows-msvc", "runs_on": "windows-latest-8-cores"},')
+          matrix+=('{"platform": "linux", "target": "x86_64-unknown-linux-gnu", "runs_on": "ubuntu-latest-16-cores", container: {"image": "rerunio/ci_docker:0.5"}}')
```

Comment:
This is a little different from how it normally looks:

https://github.com/rust-windowing/winit/blob/master/.github/workflows/ci.yml#L22-L43

Reply:
let's try this format instead, looks cleaner

Reply:
Yeah, now that it's not dynamically generated based on an env-variable, seems like we can clean it up.

Reply:
That's also something we might want to punt on though since this matrix-CI-builder mechanism has been working so far and restructuring the job graph is a good way to end up debugging job-graphs over the weekend.

Reply:
.. in a separate PR!